extension-manifest-v2/src/modules/autoconsent.js

```diff
@@ -15,41 +15,42 @@ import globals from '../classes/Globals';
 import conf from '../classes/Conf';
 
 async function initialize(msg, tabId, frameId) {
-	const url = new URL(msg.url);
 	const {
 		enable_autoconsent,
 		autoconsent_whitelist,
 		autoconsent_blacklist,
 		site_whitelist,
 	} = conf;
+	if (!enable_autoconsent || globals.SESSION.paused_blocking) {
+		return;
+	}
 
-	if (
-		enable_autoconsent &&
-		!globals.SESSION.paused_blocking &&
-		!autoconsent_blacklist?.includes(url.hostname) &&
-		!site_whitelist.includes(url.hostname.replace(/^www\./, ''))
-	) {
-		const optOut = autoconsent_whitelist?.some(h => url.hostname.includes(h)) ?? true;
+	const url = new URL(msg.url);
```

Comment:
maybe use tldts-experimental or cliqz/url-parser instead?

Reply:
Again, I would try to stay as close as possible to the other normalization functions, which are using URL in the end (content script and also the whitelist code, which uses parseUrl).